Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3528 @@
+            'start_time': start_time,
+            'end_time': end_time,
+            'limit': limit,
+            'offset': offset,
+        })
+
+        return self._http_request(
+            method='GET',
+            url_suffix=f'/vulnerabilities/{sha256}/computers',
+            params=params,
+        )
+
+
+''' COMMAND FUNCTIONS '''  # pylint: disable=pointless-string-statement
+
+
+def fetch_incidents(
+    client: Client,
+    last_run: Dict[str, Any],
+    first_fetch_time: str,
+    event_types: List[int] = None,
+    incident_severities: List[str] = None,
+    max_incidents_to_fetch: int = FETCH_LIMIT,
+) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Retrieves new alerts every interval (default is 1 minute).
+    Implements the logic of making sure that incidents are fetched only once.
+    By default it's invoked by XSOAR every minute.
+    It will use last_run to save the time of the last incident it processed and previous incident IDs.
+    If last_run is not provided, first_fetch_time will be used to determine when to start fetching the first time.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        last_run (Dict[str, Any]):
+            last_fetch: Time of the last processed incident.
+            previous_ids: List of incident IDs to that would not be repeated.
+        first_fetch_time (str): Determines the time of when fetching has been started.
+        event_types (List[int], optional): Event types to filter by.
+            Defaults to None.
+        incident_severities (List[str], optional): Incident severities to filter by.
+            Defaults to None.
+        max_incidents_to_fetch (int, optional): Max number of incidents to fetch in a single run.
+            Defaults to FETCH_LIMIT.
+
+    Returns:
+        Tuple[Dict[str, int], List[dict]]:
+            next_run: Contains information that will be used in the next run.
+            incidents: List of incidents that will be created in XSOAR.
+    """
+    last_fetch = last_run.get('last_fetch')
+    previous_ids = last_run.get('previous_ids', [])
+
+    # If a last fetch run doesn't exist, use the first fetch time.
+    if last_fetch is None:
+        last_fetch = first_fetch_time
+
+    last_fetch_timestamp = date_to_timestamp(last_fetch, ISO_8601_FORMAT)
+
+    response = client.event_list_request(
+        start_date=last_fetch,
+        event_types=event_types
+    )
+    items = response['data']
+
+    incidents: List[Dict[str, Any]] = []
+    incident_name = 'Cisco AMP Event ID:"{event_id}"'
+
+    for item in items:
+        # Break once the maximum number of incidents has been achieved.
+        if len(incidents) >= max_incidents_to_fetch:
+            break
+
+        # Continue if the incident severity isn't in the requested list (only if there is one).
+        severity = item.get('severity')
+
+        if incident_severities and severity not in incident_severities:
+            continue
```

Comment:
Because I'm using severity in the `if` statement and in building the incident, shouldn't I save it in an argument?

```
        # Continue if the incident severity isn't in the requested list (only if there is one).
        severity = item.get('severity')

        if incident_severities and severity not in incident_severities:
            continue

        # Continue if the incident ID has been fetched already.
        if (incident_id := str(item.get('id'))) in previous_ids:
            continue

        previous_ids.append(incident_id)

        incident_timestamp = item['timestamp'] * 1000
        incident = remove_empty_elements({
            'name': incident_name.format(
                event_id=incident_id,
            ),
            'occurred': timestamp_to_datestring(incident_timestamp),
            'rawJSON': json.dumps(item),
            'severity': XSOAR_SEVERITY_BY_AMP_SEVERITY.get(severity),
            'details': str(item.get('event_type')),
            'dbotMirrorId': incident_id,
        })
```

Though I can use the walrus operator in this line:
`if incident_severities and (severity := item.get('severity')) not in incident_severities:`

Reply:
right, I missed the reuse
the walrus looks good

Reply:
Thanks 🙂